org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/apt/model/TypeElementImpl.java

```diff
@@ -181,17 +182,26 @@ public List<? extends Element> getEnclosedElements() {
 
 	@Override
     public List<? extends RecordComponentElement> getRecordComponents() {
-		if (_binding instanceof SourceTypeBinding) {
-			SourceTypeBinding binding = (SourceTypeBinding) _binding;
+		if (this._binding instanceof SourceTypeBinding) {
```

Comment:
Looks like two "if" blocks are identical except first line, can that be extracted as `ReferenceBinding` line  to avoid copy/paste the rest of the code?

Reply:
Fixed.